core/src/sql/part.rs

```diff
@@ -235,3 +279,61 @@ impl fmt::Display for DestructurePart {
 		}
 	}
 }
+
+// ------------------------------
+
+#[revisioned(revision = 1)]
+#[derive(Clone, Debug, Eq, PartialEq, PartialOrd, Serialize, Deserialize, Hash)]
+#[cfg_attr(feature = "arbitrary", derive(arbitrary::Arbitrary))]
+#[non_exhaustive]
+pub enum Recurse {
+	Fixed(i64),
+	Range(Option<i64>, Option<i64>),
+}
```

Comment:
Ah I inherited this from the parser where Number(Int(i64)) is an i64. We can probably change all of these to something like `u32`, maybe even `u16`... I'd pick u32 to be on the safe side in case somebody increases the limit manually.

Reply:
Fixed in [8880f43](https://github.com/surrealdb/surrealdb/pull/5086/commits/8880f43ee87d1d3c26526b85b273c8d3fcd8a000)